src/DocumentDB.ChangeFeedProcessor.IntegrationTests/V3CompatibilityTests.cs

```diff
@@ -0,0 +1,226 @@
+﻿//----------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  Licensed under the MIT license.
+//----------------------------------------------------------------
+
+namespace Microsoft.Azure.Documents.ChangeFeedProcessor.IntegrationTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Documents.ChangeFeedProcessor.IntegrationTests.Utils;
+    using Microsoft.Azure.Documents.ChangeFeedProcessor.PartitionManagement;
+    using Microsoft.Azure.Documents.Client;
+    using Microsoft.Azure.Documents.Linq;
+    using Newtonsoft.Json.Linq;
+    using Xunit;
+
+    [Trait("Category", "Integration")]
+    [Collection("Integration tests")]
+    public class V3CompatibilityTests : IntegrationTest
+    {
+        public V3CompatibilityTests() : base()
+        {
+        }
+
+        /// <summary>
+        /// Default V2 schema has PartitionId, not LeaseToken.
+        /// </summary>
+        [Fact]
+        public async Task Schema_DefaultsToNoLeaseToken()
+        {
+            TestObserverFactory observerFactory = new TestObserverFactory(
+                (FeedProcessing.IChangeFeedObserverContext context, IReadOnlyList<Document> docs) =>
+                {
+                    return Task.CompletedTask;
+                });
+
+            IChangeFeedProcessor changeFeedProcessorBuilder = await new ChangeFeedProcessorBuilder()
+                    .WithObserverFactory(observerFactory)
+                    .WithHostName("smoke_test")
+                    .WithFeedCollection(this.MonitoredCollectionInfo)
+                    .WithLeaseCollection(this.LeaseCollectionInfo)
+                    .BuildAsync();
+
+            await changeFeedProcessorBuilder.StartAsync();
+            await Task.Delay(5000);
```

Comment:
This is technically OK but not great. The condition for initialized store is: (1) .info file exists; (2) .lock file doesn't exist. We can wait for this condition.

Reply:
Removed, added method that checks for the conditions